packages/datetime/src/dateRangeInput.tsx

```diff
@@ -616,30 +627,38 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
         let nextState: IDateRangeInputState = { shouldSelectAfterUpdate: false };
 
         if (inputString.length === 0) {
-            // this case will be relevant when we start showing the hovered
-            // range in the input fields. goal is to show an empty field for
-            // clarity until the mouse moves over a different date.
+            // this case will be relevant when we start showing the hovered range in the input
+            // fields. goal is to show an empty field for clarity until the mouse moves over a
+            // different date.
+            const baseState = { ...nextState, [keys.inputString]: "" };
             if (isValueControlled) {
-                nextState = { ...nextState, [keys.inputString]: "" };
+                nextState = baseState;
             } else {
-                nextState = { ...nextState, [keys.inputString]: "", [keys.selectedValue]: moment(null) };
+                nextState = { ...nextState, [keys.selectedValue]: moment(null) };
             }
             Utils.safeInvoke(this.props.onChange, this.getDateRangeForCallback(moment(null), boundary));
         } else if (this.isMomentValidAndInRange(maybeNextValue)) {
-            // note that error cases that depend on both fields (e.g.
-            // overlapping dates) should fall through into this block so that
-            // the UI can update immediately, possibly with an error message on
-            // the other field.
+            // note that error cases that depend on both fields (e.g. overlapping dates) should fall
+            // through into this block so that the UI can update immediately, possibly with an error
+            // message on the other field.
+            //
```

Comment:
please no whitespace in comments like this. line length should be enough indicator of new thought.